src/test/util/blocktools.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2019 The Unit-e developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/util/blocktools.h>
+
+#include <boost/test/unit_test.hpp>
+
+namespace blocktools {
+
+CBlockIndex *BlockIndexFake::MakeBlockIndex(const uint256 &hash, CBlockIndex *const prev) {
+  auto result = block_indexes.emplace(hash, CBlockIndex());
+  CBlockIndex *const index = &result.first->second;
+  index->phashBlock = &result.first->first;
+  index->nHeight = prev ? prev->nHeight + 1 : 0;
+  index->pprev = prev;
+  return index;
+}
+
+CBlockIndex *BlockIndexFake::Generate(const std::size_t count, const uint256 &starting_point) {
+  auto it = block_indexes.find(starting_point);
+  std::size_t height = 0;
+  CBlockIndex *const starting_index = [&]() {
```

Comment:
Do we really need a lambda here? I'd rather keep their presence at minimum and only where is actually clearer than the alternative.

Reply:
I'm using this style quite a lot to have the variable assigned to be marked as `const` / single assignment, which communicates that it's not changing down the further implementation of the algorithm.

It should yield the same code actually as without. In some languages, where blocks can are expressions, this would just be an expression, in C++ it looks a bit ugly, okay. Then again, I for one am also used to this style quite a lot from JavaScript. But yes, this is not JavaScript...

I'm undecided. I like const correctness.

Reply:
C++ Core Guidelines recommend using lambdas for complex initialization https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-lambda-init

Reply:
I got rid of the lambda in 700ab0f2926dd4b9a285816f23c942db50dc4eca